src/bindings/interface/types.i

```diff
@@ -126,3 +149,15 @@ Only meant to be included, shouldn't be passed to cmake as a module!
     $1 = %static_cast(val,$basetype);
   }
 }
+
+%typemap(in) libcellml::Annotator::Type (int val, int ecode) {
+  ecode = SWIG_AsVal(int)($input, &val);
+  if (!SWIG_IsOK(ecode)) {
+    %argument_fail(ecode, "$type", $symname, $argnum);
+  } else {
+    if (val < %static_cast($type::COMPONENT, int) || %static_cast($type::VARIABLE, int) < val) {
+      %argument_fail(ecode, "$type is not a valid value for the enumeration.", $symname, $argnum);
```

Comment:
Needs to extend to use XML as last type

Reply:
XML isn't required by the Annotator class so it should only come in when needed.